sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -932,10 +994,14 @@ def get_buffer(buffer_id):
       return pcoll_buffers[buffer_id]
 
     def get_input_coder_impl(transform_id):
-      return context.coders[
-          safe_coders[beam_fn_api_pb2.RemoteGrpcPort.FromString(
-              process_bundle_descriptor.transforms[transform_id].spec.payload).
-                      coder_id]].get_impl()
+      coder_id = beam_fn_api_pb2.RemoteGrpcPort.FromString(
+          process_bundle_descriptor.transforms[transform_id].spec.payload
+      ).coder_id
+      assert coder_id
+      if coder_id in safe_coders:
```

Comment:
An alternative would be `safe_coders.get(coder_id, coder_id)` which means look up the key (first argument) and if it doesn't exist return the second argument. This would eliminate some of the repetition of logic between the lines as well.